# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1330028723 28800
#      Thu Feb 23 12:25:23 2012 -0800
# Node ID e41e538e9d035f398096e24d0673153ee4f65953
# Parent  d17237256856236819d096ed4eafc02fd8f6d9bf
Fix TickDir handling for 2D and 3D plots (bug #35575).

* graphics.cc (update_ticklength): Set TickDir direction based
on 2D versus 3D.

diff --git a/src/graphics.cc b/src/graphics.cc
--- a/src/graphics.cc
+++ b/src/graphics.cc
@@ -4937,29 +4937,21 @@ axes::properties::update_axes_layout (vo
 }
 
 void
 axes::properties::update_ticklength (void)
 {
   bool mode2d = (((xstate > AXE_DEPTH_DIR ? 1 : 0) +
                   (ystate > AXE_DEPTH_DIR ? 1 : 0) +
                   (zstate > AXE_DEPTH_DIR ? 1 : 0)) == 2);
+
   if (tickdirmode_is ("auto"))
-  {
-    // FIXME: tickdir should be updated (code below comes
-    //        from JHandles)
-    //autoMode++;
-    //TickDir.set(mode2d ? "in" : "out", true);
-    //autoMode--;
-  }
-
-  //double ticksign = (tickdir_is ("in") ? -1 : 1);
-  double ticksign = (tickdirmode_is ("auto") ?
-                     (mode2d ? -1 : 1) :
-                     (tickdir_is ("in") ? -1 : 1));
+    tickdir.set (mode2d ? "in" : "out", true);
+
+  double ticksign = (tickdir_is ("in") ? -1 : 1);
 
   Matrix bbox = get_boundingbox (true);
   Matrix ticklen = get_ticklength ().matrix_value ();
   ticklen(0) = ticklen(0) * std::max (bbox(2), bbox(3));
   ticklen(1) = ticklen(1) * std::max (bbox(2), bbox(3));
 
   xticklen = ticksign * (mode2d ? ticklen(0) : ticklen(1));
   yticklen = ticksign * (mode2d ? ticklen(0) : ticklen(1));
